modules/nlu/src/views/entities/EntityEditor.jsx

```diff
@@ -1,30 +1,35 @@
 import React from 'react'
 import style from './style.scss'
-import { ListGroupItem, Glyphicon } from 'react-bootstrap'
+import { ListGroupItem, Glyphicon, Label, FormControl } from 'react-bootstrap'
 import _ from 'lodash'
 import { WithContext as ReactTags } from 'react-tag-input'
 import classNames from 'classnames'
 
+const DEFAULT_STATE = {
+  currentOccurence: undefined,
+  currentEntity: undefined,
+  occurenceInput: '',
+  pattern: ''
+}
+
 export default class EntityEditor extends React.Component {
   constructor(props) {
     super(props)
     this.occurenceInputRef = React.createRef()
   }
 
-  state = {
-    currentOccurence: undefined,
-    currentEntity: undefined,
-    occurenceInput: ''
-  }
+  state = { ...DEFAULT_STATE }
 
   static getDerivedStateFromProps(props, state) {
-    if (props.entity !== state.currentEntity) {
+    if (props.entity && props.entity !== state.currentEntity) {
       return {
         currentEntity: props.entity,
-        currentOccurence: props.entity && props.entity.occurences[0]
+        currentOccurence: props.entity && props.entity.occurences[0],
```

Comment:
`props.entity` already checked in above if, did you mean `props.entity.occurences`?